core/builder/src/main/java/io/quarkus/builder/ItemId.java

```diff
@@ -11,14 +11,16 @@
  */
 final class ItemId {
     private final Class<? extends BuildItem> itemType;
+    private final boolean multi;
 
     ItemId(final Class<? extends BuildItem> itemType) {
         Assert.checkNotNullParam("itemType", itemType);
         this.itemType = itemType;
+        this.multi = MultiBuildItem.class.isAssignableFrom(itemType);
     }
 
     boolean isMulti() {
-        return MultiBuildItem.class.isAssignableFrom(itemType);
```

Comment:
I don't think this change is needed here, as the check should be fast already.

Reply:
Sorry, forgot to revert this.